driver/src/main/java/org/neo4j/driver/internal/handlers/TransactionPullResponseCompletionListener.java

```diff
@@ -45,6 +45,9 @@ public void afterFailure( Throwable error )
         // always mark transaction as terminated because every error is "acknowledged" with a RESET message
         // so database forgets about the transaction after the first error
         // such transaction should not attempt to commit and can be considered as rolled back
-        tx.markTerminated( error );
+        if ( tx.isOpen() )
```

Comment:
Is there a danger that we lose the error if the tx isn't open?

Reply:
This was mainly to account for the fact that `RunResponseHandler` may mark transaction before this gets executed as part of pull response processing.
However, the approach has been changed now and this update has been reverted.
Both `NetworkSession` and `UnmanagedTransaction` are expected to let all pull handlers execute before completing the result cursors, meaning that neither commit or rollback transitions are expected before this status is marked. Transaction may be marked from both `RunResponseHandler` and `TransactionPullResponseCompletionListener`.